### PR TITLE
[13.0][IMP] sale_coupon_auto_refresh: configurable triggers

### PR DESCRIPTION
**Pull request by chienandalu**
This can allow us to work altogether with flexible triggers like the
ones we could need with sale_coupon_criteria_order_based

cc @Tecnativa TT37479

### CODE REVIEW

File: sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py [hunk 1]
```diff
@@ -17,11 +17,21 @@ class SaleCouponRefreshMixin(models.AbstractModel):
...
 
-        Hook method to be overridden if necessary
+        The method is overriden in the proper modules to set some basic triggers though
         """
-        return set()
+        additional_triggers = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("sale_coupon_auto_refresh.%s_triggers" % (self._table), "")
+            .split(",")
```
Comment (pedrobaeza) [inline sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py:34]: Maybe someone introduce the list of fields as `field1, field2, field3`

```suggestion
            .split(",").strip()
```

Comment (chienandalu) [reply sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py:34]: Done with `.replace(" ", "").split(",")` `strip()` only removes leading and trailing spaces.

Comment (pedrobaeza) [reply sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py:34]: Sorry, I tried to put the `strip()` on each element of the list.

**Review by pedrobaeza — APPROVED** [review]

File: sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py [hunk 2]
```diff
@@ -17,11 +17,22 @@ class SaleCouponRefreshMixin(models.AbstractModel):
 
     @api.model
     def _get_auto_refresh_coupons_triggers(self) -> set:
-        """Returns set of fields which trigger the recomputation
+        """Returns set of fields which trigger the recomputation. We can configure
+        additional triggers via config parameters setting them up as list of fields
+        separated by commas. For example, for sale.order, the config param key would be
+        sale_coupon_auto_refresh.sale_order_triggers and the value something like:
+        warehouse_id,carrier_id
```
Comment (SilvioC2C) [inline sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py]:
```suggestion
        """Returns set of fields which trigger the recomputation.
        
        Configure additional triggers via config parameters, setting them up
        as list of fields separated by commas. For example, for sale.order,
        the param key would be sale_coupon_auto_refresh.sale_order_triggers,
        and the value something like: "warehouse_id,carrier_id".
```

Comment (chienandalu) [reply sale_coupon_auto_refresh/models/sale_coupon_refresh_mixin.py]: Changed :)

**Review by SilvioC2C — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@chienandalu one test is failing. Can you please check?

**Comment by chienandalu** [comment]
yes, I'm trying to reproduce it in my local :confused:

**Comment by chienandalu** [comment]
Green at last

**Pull request merged**